lib/src/components/NxTag/__tests__/NxTag.test.tsx

```diff
@@ -4,124 +4,125 @@
 });
 
 describe('NxSelectableTag', function() {
-  const minimumProps = { children: 'selectable tag', selected: false, onSelect: jest.fn() };
-  const getShallowComponent = enzymeUtils.getShallowComponent(NxSelectableTag, minimumProps);
+  type PropsWithRef = SelectableProps & RefAttributes<HTMLLabelElement>;
+
+  const minimalProps = { children: 'selectable tag', selected: false, onSelect: jest.fn() },
+      quickRender = rtlRender<PropsWithRef>(NxSelectableTag, minimalProps),
+      renderEl = rtlRenderElement<PropsWithRef>(NxSelectableTag, minimalProps);
 
   it('renders an NxTag', function() {
-    expect(getShallowComponent()).toMatchSelector('ForwardRef(NxTag)');
+    const el = renderEl()!;
+    expect(el.tagName).toBe('LABEL');
+    expect(el).toHaveClass('nx-tag');
   });
 
   it('forwards a ref', function() {
     const ref = React.createRef<HTMLLabelElement>(),
-        component = mount(<><NxSelectableTag {...minimumProps} ref={ref}>Hello</NxSelectableTag></>);
-    expect(component.find('.nx-tag').getDOMNode()).toBe(ref.current);
+        renderedEl = renderEl({ ref });
+    expect(ref.current).toBe(renderedEl);
   });
 
   it('renders NxSelectableTag with the `nx-tag--selectable` class', function() {
-    expect(getShallowComponent()).toMatchSelector('.nx-tag--selectable');
+    const el = renderEl();
+    expect(el).toHaveClass('nx-tag--selectable');
   });
 
   it('renders the `nx-tag--unselected` class when not selected', function() {
-    expect(getShallowComponent()).toMatchSelector('.nx-tag--unselected');
+    const el = renderEl();
+    expect(el).toHaveClass('nx-tag--unselected');
   });
 
   it('renders the `nx-tag--selected` class when selected and appears before the icon', function() {
-    const getSelectedComponent = getShallowComponent({ selected: true, children: 'foo bar' });
-    expect(getSelectedComponent).toMatchSelector('.nx-tag--selected');
+    const el = renderEl({ selected: true, children: 'foo bar' });
+    expect(el).toHaveClass('nx-tag--selected');
   });
 
   it('renders the plus icon and action class when not selected', function() {
-    const SelectableTagIcons = function() {
-          return getShallowComponent().prop('selectedIcons');
-        },
-        icon = shallow(<SelectableTagIcons />);
-
-    expect(icon).toMatchSelector(NxFontAwesomeIcon);
-    expect(icon).toHaveProp(icon, faPlusCircle);
-    expect(icon).toHaveClassName('nx-tag__action');
+    const el = quickRender(),
+        icon = el.getByRole('img', { hidden: true });
+    expect(icon).toHaveClass('fa-plus-circle nx-tag__action');
   });
 
   it('renders the times icon when selected', function() {
-    const getSelectedComponent = enzymeUtils.getShallowComponent(NxSelectableTag,
-        { children: 'selectable tag', selected: true, onSelect: jest.fn() });
+    const el = quickRender({ selected: true }),
+        icon = el.getByRole('img', { hidden: true });
+    expect(icon).toHaveClass('fa-times-circle nx-tag__action');
+  });
```

Comment:
This all seems like impl details that should be covered in visual tests

Reply:
> This all seems like impl details that should be covered in visual tests

I removed all of the tests that checked for classes. a51ebe2fee38cc33d49fa21c337977c5e47bee55